File: 0108-convert-sorted-array-to-binary-search-tree/0108-convert-sorted-array-to-binary-search-tree.py
# Definition for a binary tree node.
# class TreeNode(object):
#     def __init__(self, val=0, left=None, right=None):
#         self.val = val
#         self.left = left
#         self.right = right
class Solution(object):
    def sortedArrayToBST(self, nums):
        """
        :type nums: List[int]
        :rtype: Optional[TreeNode]"""
        
        if not nums :
            return None

        def getbst(arr, start ,end):
            if  start  > end :
                return None
            
            mid = (start + end) //2
            head = TreeNode(arr[mid])
            head.left = getbst(arr,start,mid-1)
            head.right = getbst(arr,mid+1, end)
            
            return head
        return getbst( nums, 0 , len(nums)-1)

File: 0108-convert-sorted-array-to-binary-search-tree/0108-convert-sorted-array-to-binary-search-tree.java
/**
 * Definition for a binary tree node.
 * public class TreeNode {
 *     int val;
 *     TreeNode left;
 *     TreeNode right;
 *     TreeNode() {}
 *     TreeNode(int val) { this.val = val; }
 *     TreeNode(int val, TreeNode left, TreeNode right) {
 *         this.val = val;
 *         this.left = left;
 *         this.right = right;
 *     }
 * }
 */
class Solution {
    public TreeNode sortedArrayToBST(int[] nums) {

        if (nums.length == 0) return null;

        return getbst(nums,0,nums.length-1);
    }
    public TreeNode getbst(int arr[], int start , int end){

        if (start > end) return null;

        int middle = (start + end) / 2;
        
        TreeNode bst = new TreeNode(arr[middle]);
        bst.left = getbst(arr, start , middle-1);
        bst.right = getbst(arr, middle+1 , end);

        return bst;

    }
}